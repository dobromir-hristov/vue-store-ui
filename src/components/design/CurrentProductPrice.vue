<template>
  <current-product-price-provider v-slot="{ price, currency }">
    <product-price
      :price="price"
      :currency="currency"
    />
  </current-product-price-provider>
</template>

<script>
import ProductPrice from '@/components/design/ProductPrice'
import CurrentProductPriceProvider from '@/components/functional/CurrentProductPriceProvider'

/**
 * @module CurrentProductPrice
 */
export default {
  name: 'CurrentProductPrice',
  components: { ProductPrice, CurrentProductPriceProvider }
}
</script>
