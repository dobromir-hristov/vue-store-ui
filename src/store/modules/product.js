import { Product } from '@/models/Product'
import { ProductService } from '@/services/api/ProductService'
import productForm from './productForm'

const SET_PRODUCT = 'SET_PRODUCT'

const state = () => ({
  /** @type Product */
  product: null
})

const getters = {
  productLoaded: state => !!state.product,
  product: state => state.product,
  currency: state => state.product.currency
}

const mutations = {
  [SET_PRODUCT] (state, product) {
    state.product = new Product(product)
  }
}

const actions = {
  async fetchProduct ({ commit, dispatch }, productId) {
    const product = await ProductService.fetchProduct(productId)
    commit(SET_PRODUCT, product)
    dispatch('productForm/prepareForm', product)
  }
}

export default {
  state,
  getters,
  mutations,
  actions,
  modules: { productForm },
  namespaced: true
}
