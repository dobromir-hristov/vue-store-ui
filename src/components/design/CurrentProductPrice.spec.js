import CurrentProductPrice from './CurrentProductPrice'
import product from '@/store/modules/product'

jest.mock('@/services/api/ProductService')
/**
 * @param overrides
 * @returns {{config, wrapper: Wrapper<V> | Wrapper<Vue> | *}}
 */
const createWrapper = async (overrides) => {
  const store = {
    product
  }
  let config = _merge(createComponentMocks({ store }), overrides)

  await config.store.dispatch('product/fetchProduct', 'some-random-id')
  const wrapper = mount(CurrentProductPrice, config)
  return {
    config,
    wrapper
  }
}

describe('CurrentProductPrice', () => {
  it('renders the CurrentProductPrice', async () => {
    const { wrapper } = await createWrapper()
    expect(wrapper.text()).toContain('$37.5')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
