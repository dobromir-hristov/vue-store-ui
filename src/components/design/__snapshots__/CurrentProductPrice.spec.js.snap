// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentProductPrice renders the CurrentProductPrice 1`] = `
<div class="ProductPrice font-serif text-3xl">
  $37.5
</div>
`;
